students/drovdahl/lesson07/html_render.py

```diff
@@ -0,0 +1,182 @@
+#!/usr/bin/env python3
+
+"""
+A class-based system for rendering html.
+Ryan Drovdahl's version
+"""
+
+
+# This is the framework for the base class
+class Element(object):
+
+    # 'html' is the default tag that is only used for testing
+    tag = 'html'
+    close_tag_line = True
+    link_tag = False
+    meta_tag = False
+
+    def __init__(self, content=None, style=None, id=None, indent=4):
+        if content is None:
+            self.content = []
+        else:
+            self.content = [content]
+        self.style = style
+        self.id = id
+        self.indent = indent
+
+    def append(self, new_content):
+        self.content.append(new_content)
+
+    def open_tag(self):
+        if self.id and self.style:
+            return f'<{self.tag} id="{self.id}" style="{self.style}">'
+        elif self.style:
+            return f'<{self.tag} style="{self.style}">'
```

Comment:
And this code should become a loop over the dict, not this simple if statements...